pageserver/src/timelines.rs

```diff
@@ -105,13 +106,21 @@ fn run_initdb(conf: &'static PageServerConf, initdbpath: &Path) -> Result<()> {
 //
 fn bootstrap_timeline(
     conf: &'static PageServerConf,
-    tenantid: ZTenantId,
-    tli: ZTimelineId,
+    tenant_id: ZTenantId,
+    timeline_id: ZTimelineId,
     repo: &Repository,
 ) -> Result<Arc<Timeline>> {
-    let initdb_path = conf
-        .tenant_path(&tenantid)
-        .join(format!("tmp-timeline-{}", tli));
+    let timeline_path = conf.timeline_path(&timeline_id, &tenant_id);
+    let initdb_path = path_with_suffix_extension(
+        timeline_path.with_file_name(format!(
+            "basebackup-{}",
+            timeline_path
+                .file_name()
+                .expect("Timeline directory should have a filename")
+                .to_string_lossy()
+        )),
+        TEMP_FILE_SUFFIX,
+    );
```

Comment:
I find this hard to understand. What exactly is the resulting path and filename? Maybe add a comment with an example, or simplify. And "basebackup" seems out-of-place, that was perhaps a copy-pasto. Maybe:
```
// Construct path with temporary suffix, i.e `tenants/<tenant_id>/timelines/<timeline_id>.temp`
let initdb_path = conf.timelines_path(&tenant_id).join(format!("{timeline_id}.{TEMP_FILE_SUFFIX}"));
```

Reply:
That is actually a special directory, that's created by basebackup and then deleted.
It was originally created in the tenant directory, I moved it one level down, into `timelines/`, but there it needs something more than a `timeline_path.file_name()` which is effectively `ZTimelineId`, so I've added a prefix.

But sure, will update.